monkey_island/cc/services/config.py

```diff
@@ -794,29 +797,42 @@
     }
 }
 
+ENCRYPTED_CONFIG_ARRAYS = \
+    [
+        ['basic', 'credentials', 'exploit_password_list'],
+        ['internal', 'exploits', 'exploit_lm_hash_list'],
+        ['internal', 'exploits', 'exploit_ntlm_hash_list']
+    ]
+
 
 class ConfigService:
+    default_config = None
+
     def __init__(self):
         pass
 
     @staticmethod
-    def get_config(is_initial_config=False):
+    def get_config(is_initial_config=False, should_decrypt=True):
         config = mongo.db.config.find_one({'name': 'initial' if is_initial_config else 'newconfig'}) or {}
         for field in ('name', '_id'):
             config.pop(field, None)
+        if should_decrypt and len(config) > 0:
+            ConfigService.decrypt_config(config)
         return config
 
     @staticmethod
-    def get_config_value(config_key_as_arr, is_initial_config=False):
-        config_key = reduce(lambda x, y: x+'.'+y, config_key_as_arr)
+    def get_config_value(config_key_as_arr, is_initial_config=False, should_decrypt=True):
```

Comment:
You're inconsistent on when are you explicit/implict on `should_decrypt`.
get_config, 2 implicit true, 1 explicit false
get_config_value - all explicitly true
get_flat_config - 1 implicit true

Why? I'm not sure it really matters except I'd rather be explicit where possible, so we know when are we passing plain credentials and we can verify it's always to the monkey.

Reply:
Fixed